File: EscapeFromTrinityEngineStats/Models/ShopRecord.cs
using EscapeFromTrinityEngineStats.Models.Instances;
using EscapeFromTrinityEngineStats.Models.Relationships;

namespace EscapeFromTrinityEngineStats.Models
{
    public class ShopRecord
    {
        public int Id { get; set; }
        public int GoldEntered { get; set; }
        public virtual List<ShopAffordableCard> AffordableCards { get; set; }
        public virtual List<ShopPurchasedCard> PurchasedCards { get; set; }
        public virtual List<ShopAffordablePassive> AffordablePassives { get; set; }
        public virtual List<ShopPurchasedPassive> PurchasedPassives { get; set; }
        public int GoldSpent { get; set; }
        public bool UpgradePurchased { get; set; }
        public CardRecord? UpgradedCard { get; set; }
    }
}

File: EFTEDashboard/src/app/interfaces/shop-record.ts
import { CardRecord } from "./card-record";
import { GameVersion } from "./game-version";
import { ShopAffordableCard } from "./relationships/shop-affordable-card";
import { ShopAffordablePassive } from "./relationships/shop-affordable-passive";
import { ShopPurchasedCard } from "./relationships/shop-purchased-card";
import { ShopPurchasedPassive } from "./relationships/shop-purchased-passive";

export interface ShopRecord {
  id?: number;
  goldEntered?: number;
  versionId?: number;
    version?: GameVersion;
  affordableCards?: ShopAffordableCard[];
  purchasedCards?: ShopPurchasedCard[];
  affordablePassives?: ShopAffordablePassive[];
  purchasedPassives?: ShopPurchasedPassive[];
  goldSpent?: number;
  upgradePurchased?: boolean;
  upgradedCard?: CardRecord | null;
}

